src/libs/actions/connections/NetSuiteCommands.ts

```diff
@@ -283,6 +283,125 @@ function updateNetSuiteSubsidiary(policyID: string, newSubsidiary: SubsidiaryPar
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    connections: {
+                        netsuite: {
+                            options: {
+                                config: {
+                                    syncOptions: {
+                                        mapping: {
+                                            [mappingName]: mappingValue,
+                                            pendingFields: {
+                                                [mappingName]: null,
+                                            },
+                                        },
+                                    },
+                                    errorFields: {
+                                        [mappingName]: null,
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    connections: {
+                        netsuite: {
+                            options: {
+                                config: {
+                                    syncOptions: {
+                                        mapping: {
+                                            [mappingName]: oldMappingValue,
+                                            pendingFields: {
+                                                [mappingName]: null,
+                                            },
+                                        },
+                                    },
+                                    errorFields: {
+                                        [mappingName]: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('common.genericErrorMessage'),
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+            },
+        ],
+    };
+
+    const params = {
+        policyID,
+        mapping: mappingValue,
+    };
+
+    let commandName;
+    switch (mappingName) {
+        case 'departments':
+            commandName = WRITE_COMMANDS.UPDATE_NETSUITE_DEPARTMENTS_MAPPING;
+            break;
+        case 'classes':
+            commandName = WRITE_COMMANDS.UPDATE_NETSUITE_CLASSES_MAPPING;
+            break;
+        case 'locations':
+            commandName = WRITE_COMMANDS.UPDATE_NETSUITE_LOCATIONS_MAPPING;
+            break;
+        case 'customers':
+            commandName = WRITE_COMMANDS.UPDATE_NETSUITE_CUSTOMERS_MAPPING;
+            break;
+        case 'jobs':
+            commandName = WRITE_COMMANDS.UPDATE_NETSUITE_JOBS_MAPPING;
```

Comment:
Just a note is that we'll change these later down the line once we have a new API for this.